integration-tests/actions/vrf/vrfv2/contract_steps.go

```diff
@@ -778,28 +468,130 @@ func RequestRandomnessAndWaitForFulfillment(
+			minimumConfirmations,
+			callbackGasLimit,
+			numberOfWords,
+			randomnessRequestCountPerRequest,
+		)
+		if err != nil {
+			l.Error().Err(err).Msg(err.Error())
+			errorChannel <- err
+		}
+	}()
+	go func() {
+		fulfillmentEvents, err := WaitForRequestAndFulfillmentEvents(
+			consumer.Address(),
+			coordinator,
+			vrfKeyData,
+			subID,
+			randomWordsFulfilledEventTimeout,
+			l,
+		)
+		if err != nil {
+			l.Error().Err(err).Msg("error waiting for RandomnessRequested and RandomWordsFulfilled events")
+			errorChannel <- err
+		}
+		ch <- fulfillmentEvents
+	}()
+	for {
+		select {
+		case err := <-errorChannel:
+			return nil, err
+		case fulfillmentEvent := <-ch:
+			return fulfillmentEvent, nil
+		case <-time.After(randomWordsFulfilledEventTimeout):
+			return nil, fmt.Errorf("timeout waiting for RandomnessRequested and RandomWordsFulfilled events")
+		}
 	}
+}
 
-	fulfillmentEvents, err := WaitForRequestAndFulfillmentEvents(
+func RequestRandomnessAndWaitForRequestedEvent(
+	l zerolog.Logger,
+	consumer contracts.VRFv2LoadTestConsumer,
+	coordinator contracts.VRFCoordinatorV2,
+	subID uint64,
+	vrfKeyData *vrfcommon.VRFKeyData,
+	minimumConfirmations uint16,
+	callbackGasLimit uint32,
+	numberOfWords uint32,
+	randomnessRequestCountPerRequest uint16,
+	randomnessRequestCountPerRequestDeviation uint16,
+	randomWordsRequestedEventTimeout time.Duration,
+) (*vrf_coordinator_v2.VRFCoordinatorV2RandomWordsRequested, error) {
+	logRandRequest(
+		l,
 		consumer.Address(),
-		coordinator,
-		vrfKeyData,
+		coordinator.Address(),
 		subID,
-		randomWordsFulfilledEventTimeout,
-		l,
+		minimumConfirmations,
+		callbackGasLimit,
+		numberOfWords,
+		randomnessRequestCountPerRequest,
+		randomnessRequestCountPerRequestDeviation,
+		vrfKeyData.KeyHash,
 	)
-	return fulfillmentEvents, err
+	ch := make(chan *vrf_coordinator_v2.VRFCoordinatorV2RandomWordsRequested)
+	errorChannel := make(chan error)
+	go func() {
+		_, err := consumer.RequestRandomness(
+			vrfKeyData.KeyHash,
+			subID,
+			minimumConfirmations,
+			callbackGasLimit,
+			numberOfWords,
+			randomnessRequestCountPerRequest,
+		)
+		if err != nil {
+			l.Error().Err(err).Msg(err.Error())
+			errorChannel <- err
+		}
+	}()
+	go func() {
+		randomWordsRequestedEvent, err := coordinator.WaitForRandomWordsRequestedEvent(
+			[][32]byte{vrfKeyData.KeyHash},
+			[]uint64{subID},
+			[]common.Address{common.HexToAddress(consumer.Address())},
+			time.Minute*1,
+		)
+		if err != nil {
+			l.Error().Err(err).Msg(err.Error())
+			errorChannel <- err
+		}
+		LogRandomnessRequestedEvent(l, coordinator, randomWordsRequestedEvent)
+		if err != nil {
+			l.Error().Err(err).Msg("error waiting for RandomnessRequested events")
+			errorChannel <- err
+		}
+		ch <- randomWordsRequestedEvent
+	}()
+	for {
+		select {
+		case err := <-errorChannel:
+			return nil, err
+		case randRequestedEvent := <-ch:
+			return randRequestedEvent, nil
+		case <-time.After(randomWordsRequestedEventTimeout):
```

Comment:
Also for adding a small delta to this timeout, see explanation in one of the previous comments.